### ISSUE TITLE
Boto example deos not work and can not get object in boto3  as well

### ISSUE THREAD

**Issue by zelostech** [Alluxio/alluxio#16048]
**Alluxio Version:**
2.8

**Describe the bug**
Trying to use boto and boto3 to get object, however, there is a error of
"botocore.errorfactory.NoSuchKey: An error occurred (NoSuchKey) when calling the GetObject operation: Path "/xxxxx/xxxx" does not exist." (boto)
"botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found" (boto3)

**To Reproduce**
1. use the boto example in your page
2. use boto3 client.download_file function

**Expected behavior**
shall download smoothly

**Urgency**
Very Urgent

**Are you planning to fix it**
no

**Additional context**

**Comment by HelloHorizon** [comment]
after offline discussion with @zelostech , if use boto, conn.get_all_buckets() can not get any file.

If use boto3, if the user uses it to upload or multiple uploads, the user cannot use the download file to get the file afterward.

@zelostech do you mind providing the detail reproduce steps?

**Comment by zelostech** [comment]
no problem.

1. For boto example issue:

When I use the code example as attached, I am supposed to get the bucket list of the data in alluxio. However an empty array result is return.

![1660758515816](https://user-images.githubusercontent.com/86292394/185207879-9f25cebc-e8ff-4bb3-a0ee-dc0e7e7a4c83.png)

2. For download issues:

The download code in boto3 :

`
import boto3

cli = boto3.resource('s3',
    aws_access_key_id='root',
    aws_secret_access_key='x',
    endpoint_url='http://a3.alluxio:39999/api/v1/s3'
)
resp = cli.meta.client.download_file(x', 'x', 'x')
`
It works when I use the put request to upload file by either boto3 or aws-sdk-cpp.

However after I use the code:
`
s3 = boto3.resource('s3')
s3.meta.client.upload_file("x", "x", "x")
`
or Transfer-Manager in aws-s3-cpp, which may trigger the mulitipy-parts uploading to upload the file.

Then I can't download the file already uploaded by the download script I metioned. 

The message is : botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found" (boto3)

When I try to download based on aws-sdk-cpp, the error message still remains.

3. For performance issue:

I use alluxio-fuse in the K8S. Everything is great, the 'ls' operation runs quite fast. The only issue is the download opertaion costs a lot of time. When only 3s to download the file from s3 cluster, the cp command via allusxio-fuse takes almost of 5mins.

Moreover, when I download the file by S3 proxy of alluxio, it has a better performance, however, still costs like 2 or 3mins.

I'm not sure if this is a common issue related to integrate with K8S. Please help me to give a check.

Thanks for your help.

**Comment by HelloHorizon** [comment]
FYI @ZhuTopher it's on Alluxio 2.8.0

**Comment by ZhuTopher** [comment]
@HelloHorizon Yes I noticed, thanks for pointing that out though. Also I noticed this has the label `priority-high`, but as of right now we don't have a lot of focus on the Python `botocore` libraries. Just letting you know this may take a little bit for us to get to & debug, apologies! :pray:

One thing that jumps out to me is that you have the following credentials set:
```
cli = boto3.resource('s3', aws_access_key_id='root', aws_secret_access_key='x', endpoint_url='http://a3.alluxio:39999/api/v1/s3' )
```

Can you share the output of `bin/alluxio fs ls /` for me? If there are no top-level directories which are owned by `root` in Alluxio's filesystem then an empty array is the correct behaviour for `ListAllMyBuckets`.

**Comment by HelloHorizon** [comment]
@ZhuTopher the implementation under boto/boto3 is AWS SDK, which also calls the S3 rest API. I would summarize it as a multi-upload issue with the above info.

**Comment by sg-c** [comment]
I tried alluxio os 2.8.0 release version (as opposed to SNAPSHOT) and the result is the same ---- I cannot reproduce the issue.
